docs/upgrade.html

```diff
@@ -61,6 +61,11 @@ <h4><a id="upgrade_3_3_0" href="#upgrade_3_3_0">Upgrading to 3.3.0 from any vers
     </li>
 </ol>
 
+<p>Note that upgrading client to 3.3.0 would introduce a new Sticky Partitioner, described in
```

Comment:
nit: We usually includes these kind of info in the `Notable changes in 3.3.0` section. I would put it second in that list.

Reply:
In addition to that, I would write something like:

> The partitioner used by default for records with no keys has been improved to avoid pathological behavior when one or more brokers are slow. The new logic can affect the batching behavior, which can be tuned by via batch.size
and/or linger.ms. The previous behavior can be restored by setting partitioner.class=org.apache.kafka.clients.producer.internals.DefaultPartitioner. See KIP-794 for details.

The above is just an idea, but the point is to include a short description of how it's better, highlight the potential need for additional tuning, explain how to restore the previous behavior and include a link to the KIP. I also skipped all formatting, but it's good that links, code, etc. were used.

Reply:
Fixed